### DISCUSSION TITLE
[vcpkg upgrade] add --clean-after-build option to vcpkg upgrade

### DISCUSSION THREAD

**Discussion by Be-ing** [microsoft/vcpkg#19098] [Ideas]
**Is your feature request related to a problem? Please describe.**
`--clean-after-build` prevents a build of many packages from taking up too much storage space for `vcpkg install`. `vcpkg upgrade` may encounter this problem too.

**Proposed solution**
Implement `--clean-after-build` for `vcpkg upgrade`

**Describe alternatives you've considered**
:shrug:

**Additional context**
CI services have finite storage space available.

**Comment by JackBoosY** [comment]
Should wait for us to fix the `upgrade` command fist.

**Reply by SHIHUAMarryMe** [reply]
I also need this. there are so many *.tar.gz files in "downloads".
